agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -192,6 +193,9 @@ def wait_for_port(service_pod, container, timeout: 300.0)
     def ensure_desired_state
       debug "state of #{service_pod.name}: #{service_pod.desired_state}"
       service_container = get_container(service_pod.service_id, service_pod.instance_number)
+
+      migrate_container(service_container) if service_container && legacy_container?(service_container)
```

Comment:
Does the `PodWorker` really need to care/understand about `legacy_container?`? IMO it should just call `Migrator` to do it's magic and `Migrator` decides if things should be done or not. This feels a bit like leaking details. :)

Reply:
Yeah agreed, with the `remove_restart_policy if @service_container.autostart?` the `legacy_container?` isn't really needed anymore.